File: 121. Best Time to Buy and Sell Stock/BestTimetoBuyandSellStock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        if not prices:
            return 0
        resolution=0
        curMin=prices[0]
        for index in range(len(prices)):
            curMin=min(prices[index],curMin)
            resolution=max(prices[index]-curMin,resolution)
        return resolution
        

File: 121. Best Time to Buy and Sell Stock/BestTimetoBuyandSellStock.java
public class Solution {
    public int maxProfit(int[] prices) {
        int profit = 0;
        int min = Integer.MAX_VALUE;
        for(int i = 0; i < prices.length; i++)
        {
            profit = Math.max(prices[i] - min, profit);
            min = Math.min(prices[i], min);
        }
        return profit;
    }
}